sklearn/preprocessing/data.py

```diff
@@ -1956,3 +1967,222 @@ def transform(self, X):
+        the new count feature).
+
+    Examples
+    --------
+    Given a dataset with two features and four samples, we let the transformer
+    find the number of occurences of each data point in the dataset
+
+    >>> from sklearn.preprocessing.data import CountFeaturizer
+    >>> data = [[1, 1], [1, 1], [3, 1], [0, 0]]
+    >>> cf = CountFeaturizer()
+    >>> cf.fit_transform(data)      # doctest: +NORMALIZE_WHITESPACE
+    array([[ 1.,  1.,  2.],
+           [ 1.,  1.,  2.],
+           [ 3.,  1.,  1.],
+           [ 0.,  0.,  1.]])
+    >>> cf = CountFeaturizer(inclusion=[1])
+    >>> cf.fit_transform(data)      # doctest: +NORMALIZE_WHITESPACE
+    array([[ 1.,  1.,  3.],
+           [ 1.,  1.,  3.],
+           [ 3.,  1.,  3.],
+           [ 0.,  0.,  1.]])
+    >>> data_2 = [[1, 1]]
+    >>> cf = CountFeaturizer()
+    >>> cf.fit(data).transform(data_2)
+    array([[ 1.,  1.,  2.]])
+
+    Notice how on the second fit, we set the inclusion to [0, 1]
+    which made it only count the number of instances of the second feature
+
+    See also
+    --------
+    WIP, I will make the documentation .rst afterwards
+    """
+
+    def __init__(self, inclusion='all', removal_policy=None):
+        # removal policy currently not implemented, but the method
+        # signatures are set
+        CountFeaturizer._check_params(inclusion=inclusion,
+                                      removal_policy=removal_policy)
+        if type(inclusion) == str:
+            self.inclusion = inclusion
+        elif inclusion is None:
+            self.inclusion = np.array([])
+        else:
+            self.inclusion = np.array(inclusion)
+        if removal_policy is None or type(removal_policy) == str:
+            self.removal_policy = removal_policy
+        else:
+            self.removal_policy = np.array(removal_policy)
+        self.count_cache = {}
+
+        # the num_feature variable acts as a way to ensure that the number of
+        # columns of X going into fit() is the same as the number of columns
+        # of X going into transform()
+        self.num_features = 0
+
+    @staticmethod
+    def _check_params(inclusion=None, removal_policy=None):
+
+        if inclusion is not None and inclusion != 'all':
+            if not CountFeaturizer._valid_data_type(inclusion):
+                raise ValueError("Illegal data type in inclusion")
+
+        if removal_policy is not None and removal_policy != 'inclusion' and \
+                not CountFeaturizer._valid_data_type(removal_policy):
+            raise ValueError("Illegal data type in removal_policy")
+
+    @staticmethod
+    def _valid_data_type(type_check):
+        """
+        Defines the data types that are compatible with CountFeaturizer
+        Currently, only Python lists and numpy arrays are accepted
+        """
+        return type(type_check) == np.ndarray or type(type_check) == list \
+            or type(type_check) == set
+
+    @staticmethod
+    def _check_well_formed(X):
```

Comment:
Is there a reason not to use our usual `check_array`? We usually expect input to be an array (or sparse matrix). Please read the contributors' guide.